### PR TITLE
Rework 'diag' definition

### PR DESCRIPTION
**Pull request by anton-bondarev**

### CODE REVIEW

File: src/drivers/diag/diag.c [hunk 1]
```diff
@@ -16,22 +16,13 @@
...
-static const struct diag *cdiag = &DIAG_IMPL;
-
 static const struct termios diag_tio = {
 	.c_lflag = ICANON,
 	.c_oflag = ONLCR,
 };
 
 int diag_init(void) {
-	const struct diag *tdiag = cdiag;
+	const struct diag *tdiag = pdiag;
```
Comment (Deryugin) [inline src/drivers/diag/diag.c]: Why not use pdiag directly?

File: src/drivers/diag/diag.h [hunk 2]
```diff
@@ -71,17 +71,13 @@ extern enum diag_kbhit_ret diag_kbhit(void);
...
-		.ops = &__DIAG_OPS_NAME(__EMBUILD_MOD__), \
-       	}
-
-#define __DIAG_OPS_NAME(_modname) \
-	MACRO_CONCAT(_modname, _diag_ops)
+#define DIAG_OPS_DEF(...) \
+	static const struct diag_ops diag_ops = { __VA_ARGS__ }; \
+	static const struct diag diag = { \
+		.ops = &diag_ops, \
+	} \
```
Comment (Deryugin) [inline src/drivers/diag/diag.h]: Missed semicolon?

File: src/drivers/diag/diag.h [hunk 3]
```diff
@@ -71,17 +71,13 @@ extern enum diag_kbhit_ret diag_kbhit(void);
...
+	const struct diag *pdiag;
```
Comment (Deryugin) [inline src/drivers/diag/diag.h]: Why not initialize pdiag here?

File: src/drivers/console/vc/vga.c [hunk 4]
```diff
@@ -122,14 +122,15 @@ static void vc_diag_putc(const struct diag *diag, char ch) {
...
-const struct vc_video_diag DIAG_IMPL_NAME(__EMBUILD_MOD__) = {
+
+static const struct vc_video_diag vc_video_diag = {
 	.diag = {
 		.ops = &vc_video_diag_ops,
 	},
 	.video = &vc_vga_video.video,
 };
-
+const struct diag *pdiag = &vc_video_diag.diag;
```
Comment (Deryugin) [inline src/drivers/console/vc/vga.c]: Why not use DIAG_OPS_DEF?

Comment (anton-bondarev) [reply src/drivers/console/vc/vga.c]: there is 'video' field in struct vc_video_diag except 'diag'

Comment (Deryugin) [reply src/drivers/console/vc/vga.c]: But this structure is not used anywhere.

File: src/drivers/serial/stm32_usart/stm_usart.c [hunk 5]
```diff
@@ -200,13 +200,8 @@ static const struct uart_params diag_defparams = {
...
 };
 
-const struct uart_diag DIAG_IMPL_NAME(__EMBUILD_MOD__) = {
-		.diag = {
-			.ops = &uart_diag_ops,
-		},
-		.uart = &stm32_uart0,
-		.params = &diag_defparams,
-};
+
```
Comment (Deryugin) [inline src/drivers/serial/stm32_usart/stm_usart.c]: Double blank line

File: src/drivers/console/vc/vga.c (hunk 4 above)
Comment (anton-bondarev) [reply src/drivers/console/vc/vga.c]: Ops, thanks, fix

Comment (Deryugin) [reply src/drivers/console/vc/vga.c]: But why don't use `DIAG_OPS_DEF`?

**Pull request merged**